iothub/device/src/IotHubBaseClient.cs

```diff
@@ -558,14 +558,14 @@ private protected static ClientPipelineBuilder BuildPipeline()
             return !isFound ? default : (T)handler;
         }
 
-        internal async Task OnMessageReceivedAsync(Message message)
+        internal async Task<MessageAcknowledgement> OnMessageReceivedAsync(Message message)
         {
             if (Logging.IsEnabled)
                 Logging.Enter(this, message, nameof(OnMessageReceivedAsync));
 
             if (message == null)
```

Comment:
maybe combine message and callback being null.

Reply:
That would be a bit wonky since I'd like to log in each case what happened, so I'd rather not combine these